tests/functional/aws-node-sdk/test/object/getObj.js

```diff
@@ -0,0 +1,216 @@
+        };
+
+        /**
+        * Create bucket if is not created
+        * If the bucket is already create, we must empty it
+        */
+        bucketUtil.createOne(bucketName)
+        .then(() => {
+            generateDataSet();
+            done();
+        }).catch(() => {
+            bucketUtil.empty(bucketName).then(() => {
+                generateDataSet();
+                done();
+            });
+        });
+    });
+
+    after(done => {
+        const bucketUtil = new BucketUtility('default');
+
+        bucketUtil.empty(bucketName).then(() => {
+            done();
+            bucketUtil.deleteOne(bucketName);
+        }).catch(done);
+    });
+
+    const matrix = new testing.matrix.TestMatrix(params);
+
+    matrix.generate(['auth'], matrix => {
+        const bucketUtil = new BucketUtility('default', matrix.params.auth);
+
+        matrix.generate(['Delimiter', 'Prefix', 'MaxKeys', 'EncodingType',
+        'Bucket'], (matrix, done) => {
+            /**
+            * usual test
+            */
+            delete matrix.params.auth;
+            bucketUtil.s3.listObjects(matrix.params, (err, data) => {
+                const MaxKeys = matrix.params.MaxKeys;
+                const maxNumberOfKeys = Math.min(1200, MaxKeys);
+
+                const prefix = matrix.params.Prefix;
+                const delimiter = matrix.params.Delimiter;
+
+                const isPrefixMatch = (delimiter && prefix)
+                ? prefix.indexOf(delimiter) !== -1
+                : false;
+                const isGoodPrefix = validPrefix.indexOf(prefix) !== -1
+                || prefix === undefined;
+                if (isGoodPrefix === false) {
+                    assert.equal(err === null, true);
+                    assert.equal(data.Contents.length === 0
+                        || data.Contents === undefined, true);
+                } else if (matrix.params.MaxKeys !== undefined
+                    && isGoodPrefix && isPrefixMatch) {
+                    assert.equal(err === null, true);
+
+                    const NumberOfData = data.Contents.length;
+                    assert.equal(NumberOfData <= maxNumberOfKeys, true);
+                    assert.equal(NumberOfData > 0, true);
+                    /**
+                    * No need to implement the other test if
+                    * EncodingType is not null because we want to see
+                    * they didn't encode string.
+                    **/
+                    if (matrix.params.EncodingType === 'url') {
+                        done();
+                        return;
+                    }
+
+                    const specialCharacter = String.fromCharCode(1);
+                    const idxSpecialCharacter = data.Contents[0]
+                    .Key.indexOf(specialCharacter);
+                    assert.equal(idxSpecialCharacter !== -1, true);
+                }
+                done();
+            });
+        }, "should get objects")
```

Comment:
Yes, each variation of the matrix is display automatically by the matrix module.